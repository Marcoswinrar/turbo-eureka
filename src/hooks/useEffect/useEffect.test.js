import { fireEvent, render, screen } from "@testing-library/react"
import { act } from "react-dom/test-utils"
import Counter from "."

describe("Componente <Counter/>", () => {
  
  beforeAll(() => {
    jest.useFakeTimers()
    jest.spyOn(global, 'setInterval')
  })
  
  it("Deve renderizar componente de Counter sem problemas", () => {
    const { container } = render(<Counter />)

    expect(container).toMatchSnapshot()
    expect(screen.queryByText(/Hooks - useEffect/)).toBeInTheDocument()    
  })
  
  it("Deve iniciar o timer ao carregar o Counter",() => {
    render(<Counter/>)    
    expect(setInterval).toHaveBeenCalledTimes(1)
  })

  it("Deve incrementar o contador ao clicar em Adicionar", () => {
    render(<Counter />)
    const buttonAdd = screen.queryByText(/Adicionar/)

    act(() => {
      fireEvent.click(buttonAdd)
    })

    expect(screen.queryByText(/1/)).toBeInTheDocument()
  })
})
