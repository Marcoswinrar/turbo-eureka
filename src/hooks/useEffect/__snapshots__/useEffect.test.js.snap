// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente <Counter/> Deve renderizar componente de Counter sem problemas 1`] = `
<div>
  <div
    class="sc-bczRLJ"
  >
    <p
      class="sc-gsnTZi"
    >
      <strong>
        Estudo sobre hooks - useEffect
      </strong>
    </p>
    <p
      class="sc-gsnTZi"
    >
      0
       atualizando!
    </p>
    <p
      class="sc-gsnTZi"
    >
      O Usu√°rio clicou 
      0
       vezes no contador!
    </p>
    <button
      class="sc-dkzDqf"
    >
      Adicionar
    </button>
  </div>
</div>
`;
